e2e/nomostest/new.go

```diff
@@ -369,6 +367,11 @@ func setupTestCase(nt *NT, opts *ntopts.New) {
 	}
 
 	if *e2e.GitProvider == e2e.Local {
+		provider := nt.GitProvider.(*gitproviders.LocalProvider)
+		nt.T.Cleanup(func() {
+			provider.PortForwarder = nil
+		})
+		provider.PortForwarder = nt.portForwardGitServer()
 		InitGitRepos(nt, allRepos...)
```

Comment:
I like this suggestion, I'll see what I can make of this

Reply:
Consolidated `InitGitRepos` into the callback. Note that this had some knock-on effects, because:
- the way `ResetRepository` was written assumed that the port forward was ready, because it pushes an initial commit
- the callback assumes `RemoteRepositories` has been set, which was also handled by the `ResetRepository` setup after the port-forward setup

I was previously working around this by returning early in the callback for the first iteration via the `prevPodName == ""` check

To work around this with the consolidation I split up `ResetRepository` into two helper functions - one to `init` the repo and one to push the initial commit. This way we can set up the repositories before starting the port forward and waiting for it to be ready, and push the initial commits after the port forward is ready. I played with a few different approaches to solve this problem, and I preferred this approach as it minimizes the differences in how we handle the `LocalProvider` vs other git providers. Let me know what you think

Reply:
seems alright. 

I feel like there's more room to clean up by pushing down into the GitProvider abstraction, but we can do more cleanup later.